src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const stateHistory = [];
+
+  for (const action in actions) {
```

Comment:
It's better to use `for .. of` loop for arrays. Also then you won't need to get actionToDo, as your every action would already be what you need instead of index.